lib/modules/platform/codecommit/iam-client.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  GetUserCommand,
+  GetUserCommandOutput,
+  IAMClient,
+} from '@aws-sdk/client-iam';
+import type { Credentials } from '@aws-sdk/types';
+import { regEx } from '../../../util/regex';
+
+let iam: IAMClient;
+
+export function initIamClient(
+  region: string,
+  credentials: Credentials
+): IAMClient {
+  if (!iam) {
+    iam = new IAMClient({
+      region,
+      credentials,
+    });
+  }
+  return iam;
+}
+
+const userRe = regEx(/User:\s*(?<arn>[^ ]+).*/);
+
+/**
+ * This method will throw an exception only in case we have no connection
+ * 1) there is a connection and we return user.arn.
+ * 2) there is a connection but no permission for the current user, we still get his user arn in the error message
+ * 3) there is a problem in the connection to the aws api, then throw an error with the err
+ */
+export async function getUserArn(): Promise<string> {
+  const cmd = new GetUserCommand({});
+  let res;
+  try {
+    const userRes: GetUserCommandOutput = await iam.send(cmd);
+    res = userRes?.User?.Arn;
+  } catch (err) {
+    const match = userRe.exec(err.message);
```

Comment:
we should abort renovate run when there is an early error.

Reply:
please read the big comment above:
Quoting:
> * This method will throw an exception only in case we have no connection
> * 1) there is a connection and we return user.arn.
> * 2) there is a connection but no permission for the current user, we still get his user arn in the error message
> * 3) there is a problem in the connection to the aws api, then throw an error with the err

Reply:
2 types of errors here, 
1. no permission to get user policy , which we dont really care about as far as it gives us the user 
2. couldnt connect to the aws account which is actually when we should quit.

Reply:
I still think we should abort on access denied, as this is clearly a configuration error because of missing permissions.

Reply:
its a permission that we dont require why would we cancel the run on it?

Reply:
would you rather ask the customer to add permissions to get IAM User?

Reply:
This is the only way i found to get the IAM User ARN  in order to use when filtering pull requests

Reply:
What's the impact if we don't have permission? e.g. we do filtering on client side instead of server side? Or more API requests, or something else?

Reply:
i still think we should have this permission for renovate and don't rely on some crude error parsing which can fail in future when AWS changes the error.

Reply:
Resolution: document that IAM permissions are recommended, warn if they are not present, but otherwise continue